src/world/World.php

```diff
@@ -1977,14 +2027,20 @@ public function getBlockAt(int $x, int $y, int $z, bool $cached = true, bool $ad
 			$chunk = $this->chunks[$chunkHash] ?? null;
 			if($chunk !== null){
 				$block = $this->blockStateRegistry->fromStateId($chunk->getBlockStateId($x & Chunk::COORD_MASK, $y, $z & Chunk::COORD_MASK));
+				$displacedBlock = $this->blockStateRegistry->fromStateId($chunk->getDisplacedBlockStateId($x & Chunk::COORD_MASK, $y, $z & Chunk::COORD_MASK));
```

Comment:
This is going to hurt performance in the majority case, which will be air / second layer not allocated